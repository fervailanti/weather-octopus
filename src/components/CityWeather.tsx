import React, { FC, ReactElement } from 'react'
import { CityType } from '../helpers/types'
import Title from './Title'

const CityWeather: FC<CityType> = ({ name, country, time, weather }): ReactElement => (
  <div className="flex justify-between w-full">
    <div className="h-full flex flex-col justify-between">
      <div className="flex flex-col">
        <Title text={{ main: name, side: country }} size="medium" />
        <p className="text-white font-thin mt-2">
          As of <span className="font-normal">{time}</span>
        </p>
      </div>
      <div className="flex flex-col">
        <p className="text-white text-6xl font-semibold">{weather.temp.current}ยบ</p>
        <p className="text-white font-thin">{weather.description}</p>
      </div>
    </div>
    <div className="flex flex-col items-center self-end">
      <img src={weather.icon} className="h-20" alt="weather-icon" />
      <p className="text-white mt-4 font-thin text-xl">
        <span className="font-semibold">{weather.temp.min}ยบ</span> /{' '}
        <span className="font-semibold">{weather.temp.max}ยบ</span>
      </p>
    </div>
  </div>
)

export default CityWeather
