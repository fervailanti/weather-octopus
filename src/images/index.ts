const req = (route: string) => require(`../images/gallery/${route}`)

export const images: any = {
  common: {
    octopus: req('common/octopus.png'),
    reload: req('common/reload.svg')
  },
  cities: {
    newYork: req('cities/new_york.jpg'),
    rome: req('cities/rome.jpg'),
    madrid: req('cities/madrid.jpg'),
    tokyo: req('cities/tokyo.jpg'),
    paris: req('cities/paris.jpg'),
    berlin: req('cities/berlin.jpg'),
  },
  weather: {
    icons: {
      sun: req('icons/sun.svg'),
      moon: req('icons/moon.svg'),
      sunCloudy: req('icons/sun-cloudy.svg'),
      moonCloudy: req('icons/moon-cloudy.svg'),
      clouds: req('icons/clouds.svg'),
      wind: req('icons/wind.svg'),
      rain: req('icons/rain.svg'),
      showerRain: req('icons/shower-rain.svg'),
      sunRain: req('icons/sun-rain.svg'),
      moonRain: req('icons/moon-rain.svg'),
      storm: req('icons/storm.svg'),
      snow: req('icons/snow.svg'),
      mist: req('icons/mist.svg'),
      forecast: req('icons/forecast.svg')
    }
  }
}