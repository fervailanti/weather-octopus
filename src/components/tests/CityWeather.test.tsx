import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { images } from '../../images'
import CityWeather from '../CityWeather'

describe('CityWeather Component', () => {
  it('Should render correctly', () => {
    const city = render(
      <CityWeather
        name="New York"
        country="USA"
        time="3:25 PM"
        weather={{
          icon: images.weather.icons.sun,
          description: 'clear sky',
          temp: { current: 14, min: 12, max: 15 }
        }}
      />
    )
    expect(city.container).toMatchSnapshot()
    expect(city.container).toHaveTextContent('New York, USA')
    expect(city.container).toHaveTextContent('As of 3:25 PM')
    expect(city.container).toHaveTextContent('12ยบ / 15ยบ')
    expect(city.container).toHaveTextContent('14ยบ')
  })
})
