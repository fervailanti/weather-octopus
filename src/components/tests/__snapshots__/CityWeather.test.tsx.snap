// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWeather Component Should render correctly 1`] = `
<div>
  <div
    class="flex justify-between w-full"
  >
    <div
      class="h-full flex flex-col justify-between"
    >
      <div
        class="flex flex-col"
      >
        <h2
          class="text-white"
        >
          <span
            class="font-semibold text-2xl"
          >
            New York
          </span>
          <span
            class="text-lg"
          >
            , 
            USA
          </span>
        </h2>
        <p
          class="text-white font-thin mt-2"
        >
          As of 
          <span
            class="font-normal"
          >
            3:25 PM
          </span>
        </p>
      </div>
      <div
        class="flex flex-col"
      >
        <p
          class="text-white text-6xl font-semibold"
        >
          14
          ยบ
        </p>
        <p
          class="text-white font-thin"
        >
          clear sky
        </p>
      </div>
    </div>
    <div
      class="flex flex-col items-center self-end"
    >
      <img
        alt="weather-icon"
        class="h-20"
        src="[object Object]"
      />
      <p
        class="text-white mt-4 font-thin text-xl"
      >
        <span
          class="font-semibold"
        >
          12
          ยบ
        </span>
         /
         
        <span
          class="font-semibold"
        >
          15
          ยบ
        </span>
      </p>
    </div>
  </div>
</div>
`;
